include/swift/Frontend/FrontendOptions.h

```diff
@@ -246,7 +246,13 @@ class FrontendOptions {
   };
 
   /// Compare the symbols in the IR against the TBD file we would generate.
-  TBDValidationMode ValidateTBDAgainstIR = TBDValidationMode::None;
+  TBDValidationMode ValidateTBDAgainstIR =
+#if !defined(NDEBUG) && defined(__APPLE__)
```

Comment:
Should this be a target check instead, in case someone builds a cross-compiler on macOS?

Reply:
I couldn't tell if this was a target or host check, and if it wasn't, couldn't work out the target check. Tips?

Reply:
Target checks are always performed at run time, and since the target isn't stored in FrontendOptions itself you'd have to do it at a bit higher level. There's not a super clean way to do this in the argument parsing in the frontend, though, because frontend options are processed first, before even the `-target` option. (Check out CompilerInvocation::parseArgs.)

One way to deal with this could be to make the field Optional, and then fall back at the use site based on the target.

Reply:
Oh, of course.